yt_dlp/extractor/bluesky.py

```diff
@@ -0,0 +1,362 @@
+            }
+        elif quoted_type == 'app.bsky.embed.external':
+            return self.url_result(traverse_obj(quoted_media, ('external', 'uri')))
+        elif quoted_type == 'app.bsky.embed.video':
+            formats, subs = self._extract_m3u8_formats_and_subtitles(
+                quoted_media.get('playlist'), video_id, 'mp4', 'm3u8_native', m3u8_id='hls', fatal=False,
+                note='Downloading m3u8 information', errnote='Unable to download m3u8 information')
+            if blob_cid := quoted_media.get('cid'):
+                quoted_did = traverse_obj(quoted_post, ('author', 'did'))
+                quoted_embed = traverse_obj(quoted_post, ('value', 'embed', ('media', None)), get_all=False)
+                endpoint = self.get_service_endpoint(quoted_did, video_id)
+                formats.append({
+                    'format_id': 'blob',
+                    'url': f'{endpoint}/xrpc/com.atproto.sync.getBlob?did={quoted_did}&cid={blob_cid}',
+                    **traverse_obj(quoted_embed, {
+                        'ext': ('video', 'mimeType', {mimetype2ext}),
+                        'width': ('aspectRatio', 'width', {int_or_none}),
+                        'height': ('aspectRatio', 'height', {int_or_none}),
+                        'filesize': ('video', 'size', {int_or_none}),
+                    }),
+                })
+            video_info = {
+                'formats': formats,
+                'subtitles': subs,
+                'thumbnail': url_or_none(quoted_media.get('thumbnail')),
+                'alt_title': quoted_embed.get('alt') or quoted_media.get('alt'),
+            }
+        else:
+            self.raise_no_formats('No video could be found in this post', expected=True)
+
+        handle = traverse_obj(post, ('author', 'handle'))
+        uploader = traverse_obj(post, ('author', 'displayName')) or handle
+
+        tags = traverse_obj(post, ('record', 'langs'), default=[])
+        if label_list := post.get('labels'):
+            tags.extend(label.get('val') for label in label_list)
+
+        return {
+            'id': video_id,
+            'title': f'{uploader} on Bluesky',
+            **video_info,
+            'uploader': uploader,
+            'channel': handle,
+            'uploader_id': did,
+            'channel_id': did,
+            'uploader_url': f'https://bsky.app/profile/{handle}',
+            'channel_url': f'https://bsky.app/profile/{did}',
+            'like_count': post.get('likeCount'),
+            'repost_count': post.get('repostCount'),
+            'comment_count': post.get('replyCount'),
+            'tags': tags,
+            'age_limit': 18 if {'sexual', 'porn', 'graphic-media'}.intersection(tags) else 0,
+            '__post_extractor': self.extract_comments(meta),
+            **traverse_obj(post, {
+                'timestamp': ('record', 'createdAt', {parse_iso8601}),
+                'description': ('record', 'text'),
+            }),
+        }
```

Comment:
The remainder I would restructure to have a generic extraction method that can then be called with the separate sub paths:
```suggestion
        entries = []
        # app.bsky.embed.video.view/app.bsky.embed.external.view
        entries.extend(self._extract_videos(post, video_id))
        # app.bsky.embed.recordWithMedia.view
        entries.extend(self._extract_videos(
            post, video_id, embed_path=('embed', 'media'), record_subpath=('embed', 'media')))
        # app.bsky.embed.record.view
        if nested_post := traverse_obj(post, ('embed', 'record', ('record', None), {dict}, any)):
            entries.extend(self._extract_videos(
                nested_post, video_id, embed_path=('embeds', 0), record_path='value'))

        if len(entries) == 1:
            return entries[0]

        return self.playlist_result(entries, video_id)
```
I would also return both the embedded post and regular media of a post in a playlist if both are present.
This mirrors the twitter extractor.